File: example/ios/Classes/bridges/AipBridge.swift
import Flutter
import CShieldSDK

class AipBridge {

    // Only the cryptographic sign/verify are exposed to Flutter. Body
    // normalization, payload construction, hashing and the response
    // timestamp-window check are all done in Dart (AIPNormalizer / CShieldAIP),
    // so no fabricated URLRequest/URL is needed here.
    func handle(call: FlutterMethodCall, result: @escaping FlutterResult) {
        let args = call.arguments as? [String: Any] ?? [:]
        switch call.method {
        case "aip.sign":   sign(args: args, result: result)
        case "aip.verify": verify(args: args, result: result)
        default:
            result(FlutterMethodNotImplemented)
        }
    }

    // MARK: - Sign / verify

    private func sign(args: [String: Any], result: FlutterResult) {
        guard let payload = args["payload"] as? String else { result(invalidArg()); return }
        do {
            let signature = try AIPCore.sign(payload)
            result(signature)
        } catch {
            result(mapError(error))
        }
    }

    private func verify(args: [String: Any], result: FlutterResult) {
        guard let payload   = args["payload"] as? String,
              let signature = args["signature"] as? String else { result(invalidArg()); return }
        do {
            let valid = try AIPCore.verifySign(payload, signature: signature)
            if valid {
                result(nil)
            } else {
                result(FlutterError(code: CShieldErrorCode.aipInvalidSignature,
                                    message: "Signature verification failed", details: nil))
            }
        } catch {
            result(mapError(error))
        }
    }

    // MARK: - Helpers

    private func invalidArg() -> FlutterError {
        FlutterError(code: CShieldErrorCode.invalidArgument, message: "Missing or invalid arguments", details: nil)
    }

    private func mapError(_ error: Error) -> FlutterError {
        if let e = error as? CShieldError {
            switch e {
            case .aipMissingHeader(let msg):    return FlutterError(code: CShieldErrorCode.aipMissingHeader,    message: msg, details: nil)
            case .aipTimestampExpired(let msg): return FlutterError(code: CShieldErrorCode.aipTimestampExpired, message: msg, details: nil)
            case .aipInvalidSignature(let msg): return FlutterError(code: CShieldErrorCode.aipInvalidSignature, message: msg, details: nil)
            case .aipSigningFailed(let msg):    return FlutterError(code: CShieldErrorCode.aipSigningFailed,    message: msg, details: nil)
            case .aipDetectProxyCA(let msg):    return FlutterError(code: CShieldErrorCode.aipProxyCA,          message: msg, details: nil)
            case .unimplement(let msg):         return FlutterError(code: CShieldErrorCode.nativeError,         message: msg, details: nil)
            }
        }
        return FlutterError(code: CShieldErrorCode.nativeError, message: error.localizedDescription, details: nil)
    }
}

File: example/android/src/main/kotlin/com/cmc/c_shield_sdk/bridges/AipBridge.kt
package com.cmc.c_shield_sdk.bridges

import android.content.Context
import android.os.Handler
import android.os.Looper
import com.cmc.c_shield_sdk.CShieldErrorCode
import com.example.c_shield_sdk.aip.api.AIPCore
import com.example.c_shield_sdk.utils.CShieldException
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel.Result

class AipBridge(private val context: Context) {

    private val mainHandler = Handler(Looper.getMainLooper())

    // Only the cryptographic sign/verify are exposed to Flutter. Body
    // normalization, payload construction, hashing and the response
    // timestamp-window check are all done in Dart (AIPNormalizer / CShieldAIP),
    // so no fabricated okhttp Request/URL is needed here.
    fun handle(call: MethodCall, result: Result) {
        when (call.method) {
            "aip.sign"   -> handleSign(call, result)
            "aip.verify" -> handleVerify(call, result)
            else -> result.notImplemented()
        }
    }

    private fun handleSign(call: MethodCall, result: Result) {
        val payload = call.argument<String>("payload")
            ?: return result.error(CShieldErrorCode.INVALID_ARGUMENT, "payload required", null)

        Thread {
            try {
                val signature = AIPCore.sign(context, payload)
                mainHandler.post { result.success(signature) }
//            } catch (e: CShieldException) {
//                mainHandler.post { result.error(CShieldErrorCode.AIP_PROXY_CA, e.message, null) }
            } catch (e: Throwable) {
                mainHandler.post { result.error(CShieldErrorCode.AIP_SIGNING_FAILED, e.message, null) }
            }
        }.start()
    }

    private fun handleVerify(call: MethodCall, result: Result) {
        val payload = call.argument<String>("payload")
            ?: return result.error(CShieldErrorCode.INVALID_ARGUMENT, "payload required", null)
        val signature = call.argument<String>("signature")
            ?: return result.error(CShieldErrorCode.INVALID_ARGUMENT, "signature required", null)

        Thread {
            try {
                val valid = AIPCore.verifySign(context, payload, signature)
                mainHandler.post {
                    if (valid) result.success(null)
                    else result.error(CShieldErrorCode.AIP_INVALID_SIGNATURE, "Signature verification failed", null)
                }
//            } catch (e: CShieldException) {
//                mainHandler.post { result.error(CShieldErrorCode.AIP_PROXY_CA, e.message, null) }
            } catch (e: Throwable) {
                mainHandler.post { result.error(CShieldErrorCode.NATIVE_ERROR, e.message, null) }
            }
        }.start()
    }

}
